### PR TITLE
Feature/sup 70 app page

### PR DESCRIPTION
**Pull request by IrynaZazyba**
added functionality to show all applications and opportunity to add supply's application

### CODE REVIEW

File: frontend/src/index.css [hunk 1]
```diff
@@ -71,67 +71,81 @@ code {
...
-.nav-link.active{
+.nav-link.active {
     color: #009edd !important;
 }
 
+.app-modal {
+    min-width: 75% !important;
+}
+
+.validationError {
```
Comment (eugenmolchanov) [inline frontend/src/index.css]: use single approach for styles naming. validation-error for example

File: src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java [hunk 2]
```diff
@@ -19,6 +19,16 @@
     @Query("select app from Application app where app.number = :number")
     Optional<Application> findByNumber(@Param("number") final String number);
 
+    @Query("select app from Application app where (:flag = true " +
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java:22]: The where clause may be refactored

File: src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java [hunk 3]
```diff
@@ -19,6 +19,16 @@
...
+            "or (:flag = false and app.wayBill is null)) and :status is null or app.applicationStatus=:status")
+    Page<Application> findAll(Pageable pageable,
+                              @Param("flag") Boolean roleFlag,
+                              @Param("status") ApplicationStatus status);
+
+    @Query("select app from Application app where :status is null or app.applicationStatus=:status")
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/repository/ApplicationRepository.java:28]: Something wrong with this where clause as well.

**Review by eugenmolchanov — CHANGES_REQUESTED** [review]
Please, see the comments

**Review by eugenmolchanov — APPROVED** [review]

File: frontend/package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "prop-types": "^15.7.2",
     "react": "^17.0.1",
     "react-bootstrap": "^1.4.0",
+    "react-bootstrap-typeahead": "^5.1.2",
```
Comment (svareyko) [inline frontend/package.json:15]: mmmm, typeahead... I already forgot about that lib...

**Review by svareyko — APPROVED** [review]

**Pull request merged**